go-controller/pkg/ovn/egressip.go

```diff
@@ -1885,6 +1891,13 @@ type egressIPController struct {
 // (routing pod traffic to the egress node) and NAT objects on the egress node
 // (SNAT-ing to the egress IP).
 func (e *egressIPController) addPodEgressIPAssignment(egressIPName string, status egressipv1.EgressIPStatusItem, pod *kapi.Pod, podIPs []*net.IPNet) (err error) {
+	if config.Metrics.EnableEIPScaleMetrics {
+		start := time.Now()
+		defer func() {
+			duration := time.Since(start)
+			metrics.RecordEgressIPAssign(duration)
```

Comment:
should be also recorded in case of error?

Reply:
> should be also recorded in case of error?

I am hoping that we would have a more centralized location for counting failures, @npinaeva . In other words, the retry object logic would be my ideal place for detecting failures. So not part of the PR, I am hoping.

Reply:
I mean since it is updated in defer, it will be updated regardless of the operation result. If you don't want to count error handling time, maybe add `err == nil` check before recording the metric?

Reply:
> I mean since it is updated in defer, it will be updated regardless of the operation result. If you don't want to count error handling time, maybe add `err == nil` check before recording the metric?

ah, sorry for missing that. Let me make that change.

Reply:
done. @npinaeva PTAL

Reply:
Looks good now!